content/Developer_Program/WebAuthn_Starter_Kit/Adaptive_Multi-Factor_Authentication.adoc

```diff
@@ -0,0 +1,169 @@
+ 
+Once the user’s profile is created, the WebAuthn Starter kit will direct the user back to the first dialog. Upon entering the username, the user will be able to log into their account - the WebAuthn Starter kit will issue their client (browser or client app) an Auth token, as defined in the OpenID connect flow. Finally, on first login, the user will be presented with a list of recovery codes generated by the WebAuthn Starter Kit, used for regaining access to their account in the event they loose their WebAuthn Authenticator. Following that dialog, the user will be passed to the main profile page, and be logged in to the application proper.
+ 
+The figure below demonstrates different types of authenticators, examples of those devices, and the Adaptive Multi-Factor Authentication prompts that they will be required to complete
+ 
+The Starter Kit does not provide an example for a user creating a resident credential upon the initial account registration. Some platform authenticators, like Touch ID and Face ID, will automatically create a resident credential once registered. Other authenticators may require you to explicitly state for one to be created based on the public key options presented by the WebAuthn Server
+
+==== User Login
+[%header,cols="1,1,1,1,1,1"]
+|===
+| *WebAuthn authenticator* | *Examples* | *Provide identifier* | *Authenticate with WebAuthn credential* | *Provide U2F Password* | *Present recovery code*
+
+| FIDO2
+a| 
+* Yubikey 5 Series
+* YubiKey BIO
+* Face ID
+* Windows Hello
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+|
+|
+
+| U2F
+a| 
+* Android Biometrics
+* YubiKey 4 Series
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+|
+
+| Usernameless - FIDO2 with a resident credential
+a| 
+* YubiKey 5 Series
+|
+a|image::Images/group-4.png[]
+|
+|
+
+| Lost or misplaced credential
+a| 
+* YubiKey 5 Series
+a|image::Images/group-4.png[]
+|
+|
+a|image::Images/group-4.png[]
+|===
+
+If a username does exist, the user is directed to the login flow. The entry dialog is a simple page directing them to plug in their authenticator (if using a roaming authenticator), before proceeding. This page also has a link to the account recovery login flow.
```

Comment:
'If username exists, the user is directed..." is not obsolete. This could be a section we explain the UX if a platform authenticator has been registered for connivence.

Reply:
Yeah, we should probably introduce and explain the trusted device flow in this area. It seems to be missing from this doc